text/0035-physics.md

```diff
@@ -0,0 +1,175 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## How NetworkRigidbody works
+
+NetworkRigidbody is a very simple component. It simply sets the kinematic mode of the Rigidbody to kinematic on all clients except the controller (server in server auth mode, owner in client auth mode).
+
+This allows the physics simulation to only be run on one machine and the resulting position to be replicated by NetworkTransform. NetworkRigidbody only works in combination with NetworkTransform and the component will enforce that NetworkTransform is present.
+ 
+## How Triggers/Collision Events Work
+
+Unity has OnTriggerEnter and OnCollisionEnter and Stay and Exit events these can be placed on any MonoBehaviour on the GameObject and will be fired if the Rigidbody collides or enters a trigger.
+
+NetworkRigidbody does not change the way trigger or collision event does and does not synchronize them over the network. What this means is:
+- On the controller all physics events are triggered correctly.
+- On the other peers which just replicate the position of the Rigidbody events can be lost. A common situation where this happens is when two dynamic bodies bounce of each other.
+
+![Physics Replication](0035-physics/repl.png)
+*Example of an edge case where collision events don't get fired on all peers. Red circles indicate frames where positions are synced to client. On the server the Rigidbody collides with the wall. The interpolated client object does never collide with the wall.*
+
+If a collision or trigger event is crucial for gameplay it is recommended to manually sychronize that event with an RPC. Here is an example for how to do that:
+```cs
+public class CollisionSynchronizer : NetworkBehaviour
+{
+    public void OnCollisionEnter(Collision collision)
+    {
+        CollisionEnterClientRPC();
+    }
+
+    public void OnCollisionExit(Collision collision)
+    {
+        CollisionExitClientRPC();
+    }
+
+    [ClientRPC]
+    private void CollisionEnterClientRPC()
+    {
+        Debug.Log("Important collision entered on the server");
+    }
+
+    [ClientRPC]
+    private void CollisionExitClientRPC()
+    {
+        Debug.Log("Important collision exited on the server");
+    }
+}
+```
+*note the above works for a server authoritative use case only* <br>
+*This example does not synchronize any information about the collision. The Collision class is not network serializable and can't be sent over an RPC. If the `other` object of the Collision is a NetworkObject the id of the object could be sent in the RPC to the clients to reference the right object*
+
+## 3D and 2D Physics
+
+3D and 2d Physics work exactly the same. The only difference is that the Rigidbody2D and NetworkRigidbody2D components need to be added to the GameObject instead of the Rigidbody and NetworkRigidbody components. The `Update` simulation mode of Physics2D is supported.
+
+# Reference-level explanation
```

Comment:
might be worth putting a note for Brian when he reads this, that a lot of the bellow should also be included